edge-modules/TwinTester/ReportedPropertiesValidator.cs

```diff
@@ -133,28 +101,39 @@ async Task<TwinCollection> ValidatePropertiesFromTwinAsync(Twin receivedTwin)
                     }
                     catch (Exception e)
                     {
-                        this.Logger.LogError($"Failed to remove validated reported property id {reportedPropertyUpdate.Key} from storage: {e}");
+                        Logger.LogError(e, $"Failed to remove validated reported property id {reportedPropertyUpdate.Key} from storage.");
                         continue;
                     }
 
                     status = $"{(int)StatusCode.Success}: Successfully validated reported property update";
-                    this.Logger.LogInformation(status + $" {reportedPropertyUpdate.Key}");
+                    Logger.LogInformation(status + $" {reportedPropertyUpdate.Key}");
                 }
                 else if (this.ExceedFailureThreshold(this.twinState, reportedPropertyUpdate.Value))
                 {
                     status = $"{(int)StatusCode.ReportedPropertyUpdateNotInCloudTwin}: Failure receiving reported property update";
-                    this.Logger.LogError(status + $" for reported property update {reportedPropertyUpdate.Key}");
+                    Logger.LogInformation($"{status} for reported property update {reportedPropertyUpdate.Key}");
                 }
                 else
                 {
                     continue;
                 }
 
                 propertiesToRemoveFromTwin[reportedPropertyUpdate.Key] = null; // will later be serialized as a twin update
-                await this.CallAnalyzerToReportStatusAsync(this.analyzerClient, Settings.Current.ModuleId, status);
+                await this.CallAnalyzerToReportStatusAsync(Settings.Current.ModuleId, status);
             }
 
             return propertiesToRemoveFromTwin;
         }
+
+        bool ExceedFailureThreshold(TwinState twinState, DateTime twinUpdateTime)
+        {
+            DateTime comparisonPoint = twinUpdateTime > twinState.LastTimeOffline ? twinUpdateTime : twinState.LastTimeOffline;
+            return DateTime.UtcNow - comparisonPoint > Settings.Current.TwinUpdateFailureThreshold;
+        }
+
+        async Task CallAnalyzerToReportStatusAsync(string moduleId, string status)
```

Comment:
Should this be consistent with the other validator and call this method HandleReportStatusAsync?